File: ios/Classes/view/banner_yandex_ad/BannerYandexAdViewFactory.swift
import Flutter
import UIKit
import YandexMobileAds

class BannerYandexAdViewFactory:
    NSObject, FlutterPlatformViewFactory {
    private let eventDispatcher: AdEventDispatcher

    init(
        eventDispatcher: AdEventDispatcher
    ) {
        self.eventDispatcher = eventDispatcher
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let argsClass = BannerAdViewArgumentsFactory.fromMap(
            args: args as! [String: Any?]
        )

        return BannerYandexAdView(
                frame: frame,
                viewIdentifier: viewId,
                arguments: nil,
                argsClass: argsClass,
                eventDispatcher: BasicAdEventDispatcherFacade(
                    uid: argsClass.uid,
                    dispatcher: eventDispatcher
                )
        )
    }

    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

File: android/src/main/kotlin/ru/kardenvan/flutter_yandex_ads_plus/ads/banner_yandex_ad/BannerYandexAdViewFactory.kt
package ru.kardenvan.flutter_yandex_ads_plus.ads.banner_yandex_ad

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event_dispatcher.FlutterYandexAdsEventDispatcher
import ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event_dispatcher_facade.BasicAdEventDispatcherFacade
import ru.kardenvan.flutter_yandex_ads_plus.platform_api.model_factories.BannerAdViewArgumentsFactory

class BannerYandexAdViewFactory(private val eventDispatcher: FlutterYandexAdsEventDispatcher) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        if (args !is Map<*, *>) {
            throw Exception("Arguments parameter is not a Map. Args: $args")
        }

        val arguments = BannerAdViewArgumentsFactory.fromMap(args)

        return BannerYandexAdView(
            context,
            arguments,
            BasicAdEventDispatcherFacade(arguments.uid, eventDispatcher)
        )
    }
}
